Lib/Stack.hpp

```diff
@@ -190,7 +234,7 @@ class Stack
   template<class It>
   static Stack fromIterator(It it) {
     Stack out;
-    out.loadFromIterator(it);
+    out.moveFromIterator(std::move(it));
```

Comment:
Was it OK to change the semantics (as I assume this did) from copy to move for all the callers? 

Would it make sense to call this convenience function also moveFromIterator to stress this?

Reply:
I actually just realized that we don't need the moveFromIterator function, because if the iterator returns a non-reference value the expression `it.next()` will be a rvalue reference anyways so the `std::move` is not needed except if the iterator returns a lvalue reference in which the move is actually not what I wanted to do. So this is a kind-of-bug and i just changed it.